### PR TITLE
A Windows and Visual Studio based Visual Studio Online Build Agent VM template

### PR DESCRIPTION
**Pull request by nzthiago**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@nzthiago__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

File: visual-studio-vsobuildagent-vm/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+      "defaultValue": "Premium_LRS",
+      "allowedValues": [
+        "Premium_LRS",
+        "Standard_LRS"
+      ],
+      "metadata": {
+        "description": "Which type of storage you want to use"
+      }
+    },
+    "DeployLocation": {
```
Comment (singhkay) [inline visual-studio-vsobuildagent-vm/azuredeploy.json]: Please make this and below parameter names camelCase

File: visual-studio-vsobuildagent-vm/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+    "PoolName": {
+      "type": "string",
+      "metadata": {
+        "description": "The Visual Studio Online build agent pool for this build agent to join. Use 'Default' if you don't have a separate pool."
+      }
+    }
+  },
+
+  "variables": {
+    "Vnet01Prefix": "10.0.0.0/16",
```
Comment (singhkay) [inline visual-studio-vsobuildagent-vm/azuredeploy.json]: camelCase for variables as well

**Comment by nzthiago** [comment]
Thanks Kay, have changed the parameters and variables to lower case, please check

**Comment by singhkay** [comment]
@nzthiago Merged! Thank you for the contribution!

**Comment by qmfrederik** [comment]
Hi @nzthiago,

I tried your build template but when configuring the build agent (in InstallVSOAgent.ps1:49), a pop up windows appears asking me to log in to Visual Studio Online.

At least in my scenario, it works if I change the login parameter from `/login:$VSOUser,$VSOPass` to ` /login:$VSOUser;$VSOPass`.

Is something wrong on my side or might it be a bug in the script itself?

Thanks!

**Comment by nzthiago** [comment]
Thanks for letting me know @qmfrederik, the agent might have been updated since I worked on this template as the comma used to work. I have to update the template with a couple of changes so will look into that at the same time. Thanks again!

**Pull request merged**